readthedocs/vcs_support/backends/git.py

```diff
@@ -79,16 +78,7 @@ def clone(self):
         code, _, err = self.run('git', 'clone', '--recursive', '--quiet',
                                 self.repo_url, '.')
         if code != 0:
-            raise ProjectImportError(
-                (
-                    "Failed to get code from '{url}' (git clone): {exit}\n\n"
-                    "git clone error output: {sterr}"
-                ).format(
-                    url=self.repo_url,
-                    exit=code,
-                    sterr=err
-                )
-            )
+            raise RepositoryError
```

Comment:
I think this is too much simpler that what we had, but what happen if there is another problem different than a private repository. I mean, the error message is very generic and we are not checking a specific error code:

For example:

```
[humitos@julia:tmp]$ GIT_TERMINAL_PROMPT=0 git clone https://github.com/readthedocs/readthedocs-corporate
Clonando en 'readthedocs-corporate'...
fatal: could not read Username for 'https://github.com': terminal prompts disabled
[humitos@julia:tmp]$  echo $?
128
```

Ha, it doesn't seem to have a different error code:
```
[humitos@julia:tmp]$ LANG=C git clone git@github.com:rtfd/readthedocs
Cloning into 'readthedocs'...
ERROR: Repository not found.
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
[humitos@julia:tmp]$ echo $?
128
```

Anyway, I think this is not really important if we don't have history/logs failing at these points because something different than a private repository.

Reply:
I agree, but this would be out of scope for this work, and would be best served by surfacing the commands we execute to users. Currently, the checkout step is hidden because it uses it's own shell execution. This could be easily moved into a BuildCommand executed through the setup environment. This way, users can actually see the output in a meaningful way.

Reply:
I like your proposal of moving it inside a BuildCommand and I agree it could be done in another PR.